### PR TITLE
dev-ifcowl-converter-agent

### PR DESCRIPTION
**Pull request by qhouyee**
This agent is intended to support the Ifc2OntoBIM agent to generate the IfcOwl Aboxes from the IFC model.

### CODE REVIEW

File: Agents/IfcOwlConverterAgent/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+#==================================================================================================
+FROM maven:3.8.6-eclipse-temurin-11-focal as builder
+
+# Copy Maven settings templates and credentials
+COPY /credentials /root/credentials
+COPY /.m2 /root/.m2
+
+# Populate settings templates with credentials, repo name
+WORKDIR /root/.m2
+# (Note that | rather than / is used as the sed delimiter, since encrypted passwords can contain the former, but not the latter
```
Comment (sm453) [inline Agents/IfcOwlConverterAgent/Dockerfile:11]: "latter, but not the former"

File: Agents/IfcOwlConverterAgent/IfcOwlConverter/pom.xml [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    <groupId>uk.ac.cam.cares.jps</groupId>
+    <artifactId>ifcowlconverter</artifactId>
+    <version>1.0.0</version>
+    <packaging>war</packaging>
+
+    <!--  Project properties  -->
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <maven.compiler.release>11</maven.compiler.release>
+        <jps.base.version>1.21.1</jps.base.version>
```
Comment (sm453) [inline Agents/IfcOwlConverterAgent/IfcOwlConverter/pom.xml:16]: I guess something more recent is not possible due to dependency incompatibility?

**Review by sm453 — CHANGES_REQUESTED** [review]

File: Agents/IfcOwlConverterAgent/Dockerfile (hunk 1 above)
Comment (qhouyee) [reply Agents/IfcOwlConverterAgent/Dockerfile]: Made the changes, and did so for the other PRs.

File: Agents/IfcOwlConverterAgent/IfcOwlConverter/pom.xml (hunk 2 above)
Comment (qhouyee) [reply Agents/IfcOwlConverterAgent/IfcOwlConverter/pom.xml:16]: Could I clarify if you mean for the JPS-base lib dependency? If so, updating the jps base lib to the latest version is possible. The dependency incompatibility is only for some Apache Jena packages. This have been resolved by explicitly stating the required 4.1.0 version within this project pom file.

Comment (sm453) [reply Agents/IfcOwlConverterAgent/IfcOwlConverter/pom.xml:16]: If it doesn't break your code, please use the most recent base lib version possible.

Comment (qhouyee) [reply Agents/IfcOwlConverterAgent/IfcOwlConverter/pom.xml]: Alright, I have updated the base lib version, and tested that it works.

**Review by sm453 — APPROVED** [review]

**Review by jlai2021 — APPROVED** [review]

**Pull request merged**